script.js

```diff
@@ -1,4 +1,341 @@
+    if (computerTotalScore > 21) {
+      computerBustMessage = `Computer bust. You win twice your bet. Refresh the page to play again.`;
+      message = `${computerBustMessage} <br><br>${computerCardsMessage} <br><br> ${playerCardsMessage}`;
+      return message;
+    }
+  }
+  // compares computer's score and player's score and outputs a message saying who is the winner (here computer's total score is > 17).
+  if (playerTotalScore == computerTotalScore) {
+    var tieMessage = `It's a tie. Your money stays the same. Refresh the page to play again.`;
+    return `${tieMessage} <br><br>${message}`;
+  } else if (playerTotalScore > computerTotalScore) {
+    var playerWinMessage = `You win. You scored higher than the dealer. You win twice your bet. Refresh the page to play again.`;
+    return `${playerWinMessage} <br><br>${message}`;
+  } else if (playerTotalScore < computerTotalScore) {
+    var playerWinMessage = `You lose. You scored lower than the dealer. You lose your initial bet. Refresh the page to play again.`;
+    return `${playerWinMessage} <br><br>${message}`;
+  }
+
+  return message;
+};
+
+// a function that determines if the score is a bust or not and outputs one of two relevant messages
+var determineIfBust = function (totalScore) {
+  if (totalScore > 21) {
+    bustMessage = `You bust. Your score is over 21. You lose your initial bet. <br><br>`;
+    return bustMessage;
+  } else {
+    bustMessage = ``;
+    return bustMessage;
+  }
+};
+
+// a function that determines if a current hand has any aces and returns true if there are, and false is there are none
+var hasAceInHand = function (playerOrComputerHand) {
+  var aceInHandCounter = 0;
+  var word = "";
+  // While Loop runs through entire hand to see if there's any aces in it and returns either true or false
+  while (aceInHandCounter < playerOrComputerHand.length) {
+    word = playerOrComputerHand[aceInHandCounter].name;
+    // checks if there are aces
+    if (word == "ace") {
+      // returns true if there is at least one ace
+      return true;
+    }
+    aceInHandCounter = aceInHandCounter + 1;
+  }
+  // returns false if  no aces
+  return false;
+};
+
+// a function that takes a current hand and determines the score. it also accommodates cases where hand has aces
+var countHandScore = function (playerOrComputerHand) {
+  var countHandScoreCounter = 0;
+  var score = 0;
+  // While Loop runs through entire hand that has no aces to return total score of whole hand
+  if (hasAceInHand(playerOrComputerHand) == false) {
+    while (countHandScoreCounter < playerOrComputerHand.length) {
+      score = score + playerOrComputerHand[countHandScoreCounter].rank;
+      countHandScoreCounter = countHandScoreCounter + 1;
+    }
+    return score;
+  }
+  // While Loop runs through an entire hand that has at least one ace to return total score of whole hand
+  else if (hasAceInHand(playerOrComputerHand) == true) {
```

Comment:
hasAceInHand(playerOrComputerHand) == true.  The " ==true " is redundant .

else if (hasAceInHand(playerOrComputerHand)){

}